source/platform/file_formats/ncch.hpp

```diff
@@ -466,14 +497,45 @@ struct NCSDHeader {
             (MediaUnit32, size)
         );
     };
+    
+    struct CartHeader {
+        BOOST_HANA_DEFINE_STRUCT(CartHeader,
+            (std::array<uint8_t, 32>, exheader_sha256),
+            (uint32_t, additional_header_size),
+            (uint32_t, sector0_offset),
+            (uint64_t, partition_flags),
+            (std::array<uint64_t, 8>, partition_id_table),
+            (std::array<uint8_t, 0x30>, unknown)
+        );
+        
+        struct Tags : expected_size_tag<0xA0> {};
+    };
+    
+    struct NANDHeader {
+        BOOST_HANA_DEFINE_STRUCT(NANDHeader,
+            (std::array<uint8_t, 0x5E>, unknown),
+            (std::array<uint8_t, 0x42>, encrypted_twl_mbr)
+        );
+        
+        struct Tags : expected_size_tag<0xA0> {};
+    };
+    
+    union NCSDExtendedHeader {
+        BOOST_HANA_DEFINE_STRUCT(NCSDExtendedHeader,
```

Comment:
Does any other code use `BOOST_HANA_DEFINE_STRUCT` with unions? Seems like this should just be a plain union (without the Tags as well). If this is required to compile, I'm not sure it will actually have reliable behavior in practice.